src/commands/cmdt/generate/records.ts

```diff
@@ -107,14 +107,27 @@ export default class Insert extends SfCommand<CreateConfigs> {
 
     // find the cmdt in the inputdir.
     // loop through files and create records that match fields
-    await Promise.all(recordConfigs.map((r) => createRecord(r)));
+    await Promise.all(recordConfigs.map(createRecord));
 
     this.log(messages.getMessage('successResponse', [flags.csv, flags['output-directory']]));
 
     return recordConfigs;
   }
 }
 
+/** pass in the column name and record.  Makes sure tht the column has a non-empty value */
+const ensureNameColumnValue =
+  (nameColumn: string) =>
+  (record: Record): string => {
+    const nameColumnValue = record[nameColumn] as unknown;
+    if (typeof nameColumnValue !== 'string' || !nameColumnValue.trim().length) {
+      throw new SfError(
+        `The column specified for the "name-column" flag (${nameColumn}) must be present in every row of the CSV file.`
+      );
+    }
+    return nameColumnValue;
+  };
```

Comment:
Does this have to be a function that returns a function?  Seems like it could just be a function that takes the record and column name and it would be simpler to read.